docs: add gcp project admin guide and reference (#150)
opstrace/opstrace: docs/references/gcp.md

# GCP project configuration

> Note: For a detailed walk-through showing how to set up a fresh GCP project for Opstrace, please try our corresponding [admin guide](<fillme>).

Reviewer: ```suggestion
Note: For a detailed walk-through showing how to set up a fresh GCP project for Opstrace, please try our corresponding [admin guide](../guides/administrator/gcp-setup.md).
```

Author: done: pushed commit

## Required services (APIs)

Creating an Opstrace cluster in a GCP project requires the following APIs to be [enabled](https://cloud.google.com/service-usage/docs/enable-disable) in that project:

* Cloud DNS API
* Compute Engine API
* Kubernetes Engine API
* Cloud SQL Admin API
* Service Networking API
* Cloud Resource Manager API

> We try to keep this list up to date -- if in doubt, please contact us!

Reviewer: ```suggestion
We try to keep this list up to date—if in doubt, please contact us!
```

Author: done: pushed commit

## Required service account permissions

Creating an Opstrace cluster in a GCP project requires a service account.
That service account must have certain security roles applied in the GCP project:

* `roles/compute.networkAdmin`
* `roles/container.admin`
* `roles/editor`
* `roles/iam.securityAdmin`

If you would like to know which individual permissions are implied by any of these roles, please search for that role (e.g. `roles/container.admin`) on the [GCP IAM permissions reference page](https://cloud.google.com/iam/docs/permissions-reference).

> Note: we didn't quite arrive at fulfilling the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege) here -- this is an ongoing effort.

Reviewer: ```suggestion
Note: we didn't quite arrive at fulfilling the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege) here—this is an ongoing effort.
```

If you're ok with it, can we use an actual em dash (U+2014)?

Author: For sure, I am used to ` -- ` being processed by an actual typesetting system into a proper em dash -- which does not happen in GH-rendered MD. So: thanks for noticing :).

Reviewer: I think we have some Markdownlint rules to build in our future.  ;)

Author: done: pushed commit

> If you have specific ideas for reducing the set if privileges required, please [open an issue](https://github.com/opstrace/opstrace/issues)!

Reviewer: ```suggestion
If you have specific ideas for reducing the set of privileges required, please [open an issue](https://github.com/opstrace/opstrace/issues/new/choose)!
```

Let's link them to the issue types page.  That way they can open a bug or even choose to go to our Discussions section.

Author: done: pushed commit

Author: update: the if/of is now also fixed on the branch.
